dags/hindawi/utils.py

```diff
@@ -0,0 +1,46 @@
+import uuid
+from datetime import date, datetime
+from io import BytesIO
+from xml.etree import ElementTree
+
+from airflow.api.common import trigger_dag
+
+
+def save_file_in_s3(data, repo):
+    if bool(data):
+        date_today = date.today().strftime("%Y-%m-%d")
+        prefix = datetime.now().strftime("%Y-%m-%dT%H:%M")
+        key = f"{date_today}/{prefix}.json"
+        byte_file = BytesIO(data)
+        repo.save(key, byte_file)
+        return key
+
+
+def split_xmls(repo, key):
+    ids_and_records = []
+    file = repo.find_by_id(key)
+    xml_string = file.getvalue().decode("utf-8")
+    for tag in ElementTree.fromstring(xml_string):
+        if "ListRecords" in tag.tag:
+            for record in tag:
+                if "record" in record.tag:
+                    id = str(uuid.uuid4())
+                    ids_and_records.append(
+                        {
+                            "id": id,
+                            "record": ElementTree.tostring(record).decode("utf-8"),
+                        }
+                    )
+    return ids_and_records
+
+
+def trigger_file_processing_DAG(ids_and_records):
+    if len(ids_and_records) > 0:
```

Comment:
```
if len(ids_and_records) > 0:
    continue
```

Reply:
annnddd same

Reply:
this condition is not in the loop, so continue is not valid here